sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -1195,4 +1196,46 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils {
       .csv(Seq("10u12").toDS())
     checkAnswer(results, Row(null))
   }
+
+  test("SPARK-21289: Support line separator") {
+    Seq(false, true).foreach { multiLine =>
+      // Read
+      Seq("a,b|1,\"a\nd\"|1,f|", "a,b|1,\"a\nd\"|1,f").foreach { lines =>
+        withTempPath { path =>
+          Files.write(path.toPath, lines.getBytes(StandardCharsets.UTF_8))
+          val df = spark.read
+            .option("multiLine", multiLine)
+            .option("lineSep", "|")
```

Comment:
The test case should be a function. We can pass different separators and verify whether it returns the expected results.